libraries/rodeos/include/eosio/key_value.hpp

```diff
@@ -254,31 +254,31 @@ class kv_table {
 
    class kv_index;
 
-   class iterator {
+   class iterator_base {
    public:
       enum class status {
          iterator_ok     = 0,  // Iterator is positioned at a key-value pair
          iterator_erased = -1, // The key-value pair that the iterator used to be positioned at was erased
          iterator_end    = -2, // Iterator is out-of-bounds
       };
 
-      iterator() = default;
+      iterator_base() = default;
```

Comment:
I would prefer to have `itr == 0 && itr_stat == iterator_end`as a class invariant; i.e. whenever, you set `itr` to 0, also set `itr_stat` to `iterator_end`. This will simplify a lot of the compare code.